lib/spack/spack/build_systems/cmake.py

```diff
@@ -94,6 +94,14 @@ class CMakePackage(PackageBase):
             description='CMake build type',
             values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
 
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_INTERPROCEDURAL_OPTIMIZATION.html
+    variant('ipo', default=False,
+            description='CMake interprocedural optimization')
+    # CMAKE_INTERPROCEDURAL_OPTIMIZATION only exists for CMake >= 3.9
+    conflicts('+ipo', when='^cmake@:3.8.2',
```

Comment:
```suggestion
    conflicts('+ipo', when='^cmake@:3.8',
```
Just in case a 3.8.3 patch release comes out someday

Reply:
Does Spack consider 3.8 to consist of everything from 3.8.0 to 3.8.99999... (i.e. 3.8.x)? I saw elsewhere in the documentation suggesting using a .999 release as the endpoint, on the assumption that it will never reach that—but that documentation could be out of date.

For reference, 3.8.2 was released in 2017—I don't think Kitware does LTS/maintenance releases, focusing on the top-of-tree instead. But I do agree with the general point :).

Reply:
Honestly, our version ranges are pretty broken. The 999 stuff is because Spack doesn't support <, only <=. 3.8 should be treated as 3.8.X in most version comparisons.